### ISSUE TITLE
importccl: computed columns and expression indexes not supported for IMPORT TABLE of CSV

### ISSUE THREAD

**Issue by pbardea** [cockroachdb/cockroach#56002]
Currently, `IMPORT TABLE` of a CSV file does not work if the schema has computed columns. This is because CSVs don't have their columns named, and can't be related to the columns in the schema. Consider:

`IMPORT TABLE (a INT, b INT AS (a + c) STORED, c INT) CSV DATA ...`

Today, IMPORT expects that the data file has 3 columns, but it should really only expect 2. Furthermore, it should know to only populate non-stored columns. Neither of these limitations seem untracktable, but there does exist a workaround:

1. Create your schema
2. Use IMPORT INTO and only target the non-computed columns

Jira issue: CRDB-3599
Epic CRDB-8353

**Comment by mgartner** [comment]
Tacking on expression indexes to this issue, which are related to computed columns because they used computed columns under the hood.

**Comment by mwang1026** [comment]
Deprecating `IMPORT TABLE ... CSV` in https://github.com/cockroachdb/cockroach/issues/66732 so closing this issue. If it becomes a problem with IMPORT INTO we can create separate issue for that

**Comment by steve-taylor** [comment]
I arrived here when attempting to `IMPORT PGDUMP`. I don't think this has anything to do with CSV files.

```
> IMPORT PGDUMP 'https://example.com/dump.sql' WITH ignore_unsupported_statements;
ERROR: unimplemented: to import into a table with expression indexes, use IMPORT INTO
SQLSTATE: 0A000
HINT: You have attempted to use a feature that is not yet implemented.
See: https://go.crdb.dev/issue-v/56002/v21.2
```

I don't know what the problem is, because in the `dump.sql` file, indexes are created at the end.

Also, these error messages are next to useless. I had to turn on the most verbose level of logging (`debug5`) to see this error message. It doesn't indicate which SQL statement or which object has triggered the error, so I can only guess what needs fixing.

I'm using CockroachDB Serverless.
